lib/elixir/lib/exception.ex

```diff
@@ -930,8 +938,11 @@ defmodule TokenMissingError do
         snippet: snippet
       })
       when not is_nil(snippet) and not is_nil(column) do
-    Exception.format_file_line_column(Path.relative_to_cwd(file), line, column) <>
-      " " <> description <> "\n" <> Exception.format_snippet(snippet, line)
+    fancy =
+      :elixir_errors.fancy_exception(file, line, column, description, snippet)
+
+    location = Exception.format_file_line_column(Path.relative_to_cwd(file), line, column)
+    format_fancy(location, fancy)
```

Comment:
I think everything is set now, I'm only having this problem locally whenever I return format_fancy from this exact line:
```
  1) test properly formats errors (Kernel.CLITest)
     test/elixir/kernel/cli_test.exs:124
     ** (ArgumentError) errors were found at the given arguments:

       * 1st argument: not an iodata term

     code: assert String.starts_with?(elixir(~c"-e \"1 +\""), "** (TokenMissingError)")
     stacktrace:
       (stdlib 4.3.1.1) :binary.list_to_bin([42, 42, 32, 40, 84, 111, 107, 101, 110, 77, 105, 115, 115, 105, 110, 103, 69, 114, 114, 111, 114, 41, 32, 116, 111, 107, 101, 110, 32, 109, 105, 115, 115, 105, 110, 103, 32, 111, 110, 32, 110, 111, 102, 105, 108, 101, 58, 49, 58, 51, ...])
       test/elixir/test_helper.exs:61: PathHelpers.run_cmd/2
       test/elixir/kernel/cli_test.exs:132: (test)

  2) test properly formats errors (Kernel.CLI.RPCTest)
     test/elixir/kernel/cli_test.exs:190
     ** (ArgumentError) errors were found at the given arguments:

       * 1st argument: not an iodata term

     code: assert String.starts_with?(rpc_eval("1 +"), "** (TokenMissingError)")
     stacktrace:
       (stdlib 4.3.1.1) :binary.list_to_bin([42, 42, 32, 40, 84, 111, 107, 101, 110, 77, 105, 115, 115, 105, 110, 103, 69, 114, 114, 111, 114, 41, 32, 116, 111, 107, 101, 110, 32, 109, 105, 115, 115, 105, 110, 103, 32, 111, 110, 32, 110, 111, 102, 105, 108, 101, 58, 49, 58, 51, ...])
       test/elixir/test_helper.exs:61: PathHelpers.run_cmd/2
       test/elixir/kernel/cli_test.exs:193: (test)
```

No idea what's happening here, it seems that `:os.cmd()` is not returning a valid iolist whenever I return it?

I'll continue debugging this tomorrow, but if you have any ideas or would like to give it a fast shot I would appreciate

Reply:
You likely have a list element that is more than 255 (a maximum byte value) somewhere. I will explore this soon and merge. :)

Reply:
Oh, the issue is in the test. The test assumes (probably run_cmd) that the result won't contain Unicode codepoints but our messages now contain them. We should ask `run_cmd` to return a binary if possible, if not, use `List.to_string` for converting the list to a string.

Reply:
If unicode chars in error output turns out to be a problem later we can use a more ascii-like appearence like rust does:  
```
** (TokenMissingError) token missing on nofile:1:4:
   -> error: nofile:1:4
   |
 1 | 1 +
   |    ^
   |
   syntax error: expression is incomplete
```

It's not that pretty, but might be an alternative.

For now I'll try to update the test code to not crash on unicode values